aws/files/autoeips.py

```diff
@@ -0,0 +1,97 @@
+{% from "aws/map.jinja" import aws with context %}
+# !/usr/bin/env python
+import boto.ec2
+from boto.exception import EC2ResponseError
+import boto.utils
+import logging
+import salt.log
+ # Set up the logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger("aws:autoeips")
+logging.getLogger("requests").setLevel(logging.WARNING)
+    
+    
+class AutoEIP():
+    filter_addresses = {{ aws.eips }}
+    ec2_connection = None
+    instance_metadata = None
+   
+    def __init__(self):
+        self.instance_metadata = self.get_instance_metadata()
+        self.instance_id = self.instance_metadata.get('instance-id')
+        # Collect details about this instance
+        vpc_id = self.instance_metadata['network']['interfaces']['macs'].values()[0]['vpc-id']
+        region = self.instance_metadata['placement']['availability-zone'][:-1]
+    
+        try:
+            self.ec2_connection = boto.ec2.connect_to_region(region)
+        except Exception as e:
+          # This prints a user-friendly error with stacktrace
+          print("Error getting EC2 conection: {}".format(e))
+          return None
+            
+    def update_association(self, force=False):
+        instance_associations = self.get_instance_association()
+        if len(instance_associations) < 1 or force:
+            print("Associating with any available eips")
+            filtered_eips = self.get_unassociated_eips()
+            self.associate_eip(filtered_eips)
+        else:
+            print("Already associated with EIP: {}".format(instance_associations[0]))
+        
+    def get_instance_association(self):
+        instance_id =self.instance_metadata.get('instance-id', None)
+        return self.ec2_connection.get_all_addresses(filters={'instance-id': instance_id})
+    
+    
+    def get_instance_metadata(self):
+        """
+        Prints a mapping of private ip addresses to private dns names
+        """
+        try:
+            instance_metadata = boto.utils.get_instance_metadata(timeout=5, num_retries=2)
+            return instance_metadata
+        except Exception as e:
```

Comment:
It'd be cleaner if a single try/except happens in the update_association method, then all other exceptions can bubble up to that place.